lib/lita/handlers/artifactory.rb

```diff
@@ -92,6 +114,35 @@ def promote(response)
         response.reply reply_msg
       end
 
+      def push(response)
+        ruby_gem      = response.args[2]
+        version       = response.args[3]
+        human_name    = "#{ruby_gem} gem version #{version}"
+        gem_source    = "#{config.endpoint}/api/gems/gems-local/"
+        missing_gem   = false
+
+        Dir.mktmpdir do |dir|
+          Dir.chdir(dir) do
+            %w{ruby universal-mingw32}.each do |platform|
+              unless system(fetch_command_for_platform(platform, ruby_gem, version, gem_source))
+                response.reply("There were errors retrieving the #{human_name} from #{gem_source}!")
+                missing_gem = true
+              end
+            end
+
+            break if missing_gem
+
+            Dir.glob("*.gem") do |gem_file|
+              if system("gem push #{gem_file} --key chef_rubygems_api_key")
```

Comment:
Yeah I had briefly looked at that and also considered mixlib-shellout, but went the simpler route because the bot really needs just a yes/no and its likely the user would then look into a local gem push as a troubleshooting matter.

However, now that another voice of reason has spoken, I do think it would provide a better user experience to get more detailed error output. I'm actually gonna take this another step and just use mixlib-shellout so both commands can report back any exceptions to the "slacker".